### PR TITLE
`@rescript/tools` package

### PR DESCRIPTION
**Pull request by aspeddro**
A binary called `restools` with two subcommands. One for document generation and another for renalyze, see README.md

The package also exposes the `Docgen` module to decode from JSON

```rescript
// Read JSON file and parse with `Js.Json.parseExn`
json->RescriptTools.Docgen.decodeFromJson
```

You can publish with the commit message `publish tools` by pushing to master.

### CODE REVIEW

**Comment by zth** [comment]
Looking great overall!

I think we can rely on the v11 variant representation here for the JSON format, and just require people to use v11. Seems like it'd clean things up a bit.

**Comment by aspeddro** [comment]
Ready for review

**Review by zth — APPROVED** [review]
Great stuff, thank you very much! I have a couple of adjustments I'd like to make but I can make them in a separate PR.

File: tools/src/Tools_Docgen.res [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  name: string,
+  docstrings: array<string>,
+  signature: string,
+  deprecated: option<string>,
+}
+
+@tag("kind")
+type detail =
+  | @as("record") Record(array<field>)
+  | @as("variant") Variant(array<constructor>)
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply tools/src/Tools_Docgen.res:19]: @zth fixed in https://github.com/rescript-lang/rescript-vscode/pull/833

**Pull request merged**